app/javascript/app/components/ndcs-map/ndcs-map.js

```diff
@@ -0,0 +1,143 @@
+import { PureComponent, createElement } from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
+import PropTypes from 'prop-types';
+import qs from 'query-string';
+
+import paths from 'app/data/world-50m-paths';
+import Component from './ndcs-map-component';
+import {
+  getCategories,
+  getIndicators,
+  getSelectedCategory,
+  getSelectedIndicator
+} from './ndcs-map-selectors';
+
+const mapStateToProps = (state, { location }) => {
+  const { ndcs } = state;
+  const search = qs.parse(location.search);
+  const ndcsWithRouter = {
+    ...ndcs,
+    category: search.category,
+    indicator: search.indicator
+  };
+
+  return {
+    categories: getCategories(ndcs),
+    indicators: getIndicators(ndcs),
+    selectedCategory: getSelectedCategory(ndcsWithRouter),
+    selectedIndicator: getSelectedIndicator(ndcsWithRouter)
+  };
+};
+
+const countryStyles = {
+  default: {
+    fill: '#ECEFF1',
+    fillOpacity: 0.3,
+    stroke: '#396d90',
+    strokeWidth: 1,
+    outline: 'none'
+  },
+  hover: {
+    fill: '#ffc735',
+    stroke: '#396d90',
+    strokeWidth: 1,
+    outline: 'none'
+  },
+  pressed: {
+    fill: '#ffc735',
+    stroke: '#396d90',
+    strokeWidth: 1,
+    outline: 'none'
+  }
+};
+
+const getChoroplethColor = (vis) => {
+  switch (vis) {
+    case 1:
+      return '#d54d60';
+    case 2:
+      return '#eebc8f';
+    case 3:
+      return '#fee08d';
+    case 4:
+      return '#7aabd3';
+    default:
+      return '#25597c';
+  }
+};
+
+class NDCMapContainer extends PureComponent {
+  handleCountryClick = (geography) => {
+    this.props.history.push(`ndcs/country/${geography.id}`);
```

Comment:
I think this needs to be /ndcs.... otherwise if you click twice it will become /ndcs.../ndcs...

Reply:
Good catch! changed and merged